client/state/posts/utils.js

```diff
@@ -91,3 +91,45 @@ export function mergeIgnoringArrays( object, ...sources ) {
 		}
 	} );
 }
+
+/**
+ * Takes existing term post edits and updates the `terms_by_id` attribute
+ *
+ * @param  {Object}    post  object of post edits
+ * @return {Object}          normalized post edits
+ */
+export function getTermIdsFromEdits( post ) {
+	if ( ! post || ! post.terms ) {
+		return post;
+	}
+
+	// Skip "default" taxonomies until legacy token-field and category-selector are removed
+	let taxonomies = omit( post.terms, [ 'post_tag', 'category' ] );
+
+	// Filter taxonomies that are set as arrays ( i.e. tags )
+	// This can be detected by an array of strings vs an array of objects
+	taxonomies = reduce( taxonomies, ( prev, taxonomyTerms, taxonomyName ) => {
+		// Ensures we are working with an array
+		const termsArray = map( taxonomyTerms );
+		if ( termsArray && termsArray.length && ! isObject( termsArray[ 0 ] ) ) {
```

Comment:
FYI, `isObject` is probably not what you wanted:

```
wp-calypso (add/term-tree-selector/add-term) $ node
> var isObject = require( 'lodash/isObject' );
undefined
> isObject( [] );
true
```

Instead, there's [`_.isPlainObject`](https://lodash.com/docs#isPlainObject)